mm2src/mm2_bitcoin/keys/src/segwitaddress.rs

```diff
@@ -150,10 +153,15 @@ impl FromStr for SegwitAddress {
         };
         // decode as bech32, should use Variant if Bech32m is used alongside Bech32
         // The improved Bech32m variant described in [BIP-0350](https://github.com/bitcoin/bips/blob/master/bip-0350.mediawiki)
-        let (_, payload, _) = bech32::decode(s)?;
+        let (_, payload, variant) = bech32::decode(s)?;
         if payload.is_empty() {
             return Err(Error::EmptyBech32Payload);
         }
+        match variant {
+            bech32::Variant::Bech32 => (),
+            bech32::Variant::Bech32m => return Err(Error::UnsupportedAddressVariant("Bech32m".into())),
+            // Important: If a new variant is added we should return an error until we support the new variant
+        }
 
         // Get the script version and program (converted from 5-bit to 8-bit)
         let (version, program): (bech32::u5, Vec<u8>) = {
```

Comment:
Do we also need to check if `version` is 0? I'm not sure if we don't need to support other versions, but we could reject `version=1` at least as it's used for taproot addresses.
@shamardy is `version=1` used for other purposes other than `taproot`?

Reply:
Found this problem already and fixed it in the latest commit :)
It's not really needed after the checksum fix but added it nonetheless to cover all bases. This is also what @caglaryucekaya mentioned in today's call